### PR TITLE
refactor: reformat and optimise colour methods

### PR DESCRIPTION
**Pull request by TomDotBat**
Just performing a well needed cleanup of the colour helpers file, includes several optimisations using scopes and localisation. Also removed some repeat functionality on the colour meta table by referencing PIXEL.

### CODE REVIEW

File: lua/pixelui/core/cl_color.lua [hunk 1]
```diff
@@ -16,27 +16,10 @@ You should have received a copy of the GNU General Public License
...
-end
-
-function PIXEL.OffsetColor(col, offset)
-    return Color(col.r + offset, col.g + offset, col.b + offset)
-end
+PIXEL.HSLToColor = HSLToColor
+PIXEL.ColorToHSL = ColorToHSL
 
-function PIXEL.Hue2RGB(p, q, t)
+function PIXEL.HueToRGB(p, q, t)
```
Comment (NoSharp) [inline lua/pixelui/core/cl_color.lua:22]: Document what "p,q,t" is, although it's relative to the orginal algorithm, it should be documented.

File: lua/pixelui/core/cl_color.lua [hunk 2]
```diff
@@ -45,96 +28,85 @@ function PIXEL.Hue2RGB(p, q, t)
...
+    local format = string.format
+    function PIXEL.DecToHex(dec, zeros)
+        return format("%0" .. (zeros or 2) .. "x", dec)
     end
-    return h * .16667, s, l
-end
 
-function PIXEL.DecToHex(d, zeros)
-    return string.format("%0" .. (zeros or 2) .. "x", d)
+    local max = math.max
```
Comment (NoSharp) [inline lua/pixelui/core/cl_color.lua:46]: I'm not sure if this is an actual benefit, as the scope here is already extremely small. Just put it in a do ... end around the function.

File: lua/pixelui/core/cl_color.lua [hunk 3]
```diff
@@ -45,96 +28,85 @@ function PIXEL.Hue2RGB(p, q, t)
...
+end
 
-    lastUpdate = time
-    lastCol = HSVToColor((time * 50) % 360, 1, 1)
+function PIXEL.LerpColor(t, from, to)
+    return createColor(from.r, from.g, from.b, from.a):Lerp(t, to)
+end
 
-    return lastCol
+function PIXEL.IsColorEqualTo(from, to)
```
Comment (NoSharp) [inline lua/pixelui/core/cl_color.lua:102]: This looks dreadful, would it not be better to store this as a number, or some sort of hash then compare the hash? Four different comparissions will be slow if called frequently.

**Review by NoSharp — CHANGES_REQUESTED** [review]

File: lua/pixelui/core/cl_color.lua (hunk 3 above)
Comment (TomDotBat) [reply lua/pixelui/core/cl_color.lua:102]: I can't imagine any use cases of this in a draw hook, your suggestion seems a little over complicated for something that's likely rarely going to be called.

File: lua/pixelui/core/cl_color.lua (hunk 1 above)
Comment (TomDotBat) [reply lua/pixelui/core/cl_color.lua]: The method was only ever used internally by PIXEL UI, so I've removed it now it has no use instead.

**Pull request merged**